src/components/FileViewer/FileContextMenu.tsx

```diff
@@ -116,15 +119,36 @@ const FileContextMenu = forwardRef(
             }}
             disabled={file === null}
           />
-          <ContextMenuButton
-            className="w-full whitespace-nowrap rounded-none bg-gray-900 px-2.5 text-small font-medium"
-            label="Unzip"
-            onClick={() => {
-              unzipFile(file as ClientFile);
-              setShowContextMenu(false);
-            }}
-            disabled={file === null}
-          />
+          {((file === null || !(zipFileExtensions.includes(file.name.split('.').pop()!)) ) ? // if file name is null or file is not zip file
+            null : (<>
+            <ContextMenuButton
+              className="w-full whitespace-nowrap rounded-none bg-gray-900 px-2.5 text-small font-medium"
+              label="Unzip here"
+              onClick={() => {
+                unzipFile(file as ClientFile, "Normal");
+                setShowContextMenu(false);
+              }}
+              disabled={file === null}
+            />
+            <ContextMenuButton
+              className="w-full whitespace-nowrap rounded-none bg-gray-900 px-2.5 text-small font-medium"
+              label="Unzip here (smart)"
+              onClick={() => {
+                unzipFile(file as ClientFile, "Smart");
+                setShowContextMenu(false);
+              }}
+              disabled={file === null}
+            />
+            <ContextMenuButton
+              className="w-full whitespace-nowrap rounded-none bg-gray-900 px-2.5 text-small font-medium"
+              label={file === null ? '' : 'Unzip to ' + file.name}
```

Comment:
It would be nice to make this ternary more readable
We probably don't need to check `file === null` explicitly, since `null` is falsy and any valid file is truthy.
Maybe something like `file ? 'Unzip to' + file.name : ''` will work?